api/tests/app/test_camera_metrics.py

```diff
@@ -0,0 +1,903 @@
+                'weeks': ['2012-04-12 2012-04-15', '2012-04-16 2012-04-22', '2012-04-23 2012-04-29',
+                          '2012-04-30 2012-05-06', '2012-05-07 2012-05-13', '2012-05-14 2012-05-18']
+            }),
+            ("face-mask-detections", {
+                'no_face': [0, 0, 0, 0, 0, 0], 'face_with_mask': [0, 0, 0, 0, 0, 0],
+                'face_without_mask': [0, 0, 0, 0, 0, 0],
+                'weeks': ['2012-04-12 2012-04-15', '2012-04-16 2012-04-22', '2012-04-23 2012-04-29',
+                          '2012-04-30 2012-05-06', '2012-05-07 2012-05-13', '2012-05-14 2012-05-18']
+            })
+        ]
+    )
+    def test_try_get_a_weekly_report_non_existent_dates(self, config_rollback_create_cameras, reports_simulation,
+                                                        metric, expected):
+        camera, camera_2, client, config_sample_path = config_rollback_create_cameras
+        camera_id = camera["id"]
+
+        response = client.get(
+            f"/metrics/cameras/{metric}/weekly?cameras={camera_id}&from_date=2012-04-12&to_date=2012-05-18")
+
+        # TODO: IF THIS BEHAVIOUR IS RIGHT
+        assert response.status_code == 200
+        assert response.json() == expected
+
+    @pytest.mark.parametrize(
+        "metric",
+        ["social-distancing", "face-mask-detections"]
+    )
+    def test_try_get_a_weekly_report_invalid_range_of_dates(self, config_rollback_create_cameras, reports_simulation,
+                                                            metric):
+        """from_date is after to_date"""
+        camera, camera_2, client, config_sample_path = config_rollback_create_cameras
+        camera_id = camera["id"]
+
+        response = client.get(
+            f"/metrics/cameras/{metric}/weekly?cameras={camera_id}&from_date=2020-09-25&to_date=2020-09-18")
+
+        assert response.status_code == 400
+
+    @pytest.mark.parametrize(
+        "metric,expected",
+        [
+            ("social-distancing", {
+                'detected_objects': [0, 654], 'no_infringement': [0, 550],
+                'low_infringement': [0, 38],
+                'high_infringement': [0, 44], 'critical_infringement': [0, 22],
+                'weeks': ['2020-09-20 2020-09-20', '2020-09-21 2020-09-23']
+            }),
+            ("face-mask-detections", {
+                'no_face': [0, 72], 'face_with_mask': [0, 482], 'face_without_mask': [0, 112],
+                'weeks': ['2020-09-20 2020-09-20', '2020-09-21 2020-09-23']
+            })
+        ]
+    )
+    def test_try_get_a_weekly_report_no_id(self, config_rollback_create_cameras, reports_simulation, metric, expected):
+        camera, camera_2, client, config_sample_path = config_rollback_create_cameras
+
+        response = client.get(
+            f"/metrics/cameras/{metric}/weekly?from_date=2020-09-20&to_date=2020-09-23")
+
+        # Todo: Ask for this behaviour. Apparently first it looks for every ID that has a report.
```

Comment:
This should be aggregating the results  for all cameras, although I think you could include a set of cameras, but maybe by default it takes all